processing/src/test/java/org/apache/druid/segment/TestIndex.java

```diff
@@ -175,6 +196,17 @@ public static IncrementalIndex getNoRollupIncrementalTestIndex()
     return noRollupRealtimeIndex = makeRealtimeIndex("druid.sample.numeric.tsv", false);
   }
 
+  public static IncrementalIndex getNoBitmapIncrementalTestIndex()
+  {
+    synchronized (log) {
+      if (noBitmapRealtimeIndex != null) {
+        return noBitmapRealtimeIndex;
+      }
+    }
```

Comment:
I know this synchronization pattern was before the change, I'm just confused what it tries to accomplish? ie, if `noBitmapRealtimeIndex` is not created, then all threads can initialize `noBitmapRealtimeIndex` simultaneously, doesn't it look strange?

Reply:
Good point, I wasn't much attention, will fix.

Reply:
Updated to use `Suppliers.memoize` instead. I looked into it, and this code is quite old! It appears to be from the [PR that added timeseries queries to Druid!](https://github.com/apache/incubator-druid/commit/499263386caed48a42fa26c72582fc97f0e64cb9#diff-1bfae2e7948227908d76a8a385844997R199) Thanks for review :+1: